src/loki/trace_route_action.cc

```diff
@@ -177,7 +177,9 @@ namespace valhalla {
     void loki_worker_t::locations_from_shape(rapidjson::Document& request) {
       std::vector<Location> locations{shape.front(), shape.back()};
       locations.front().heading_ = std::round(PointLL::HeadingAlongPolyline(shape, 30.f));
+      locations.front().node_snap_tolerance_ = 0.f;
       locations.back().heading_ = std::round(PointLL::HeadingAtEndOfPolyline(shape, 30.f));
+      locations.back().node_snap_tolerance_ = 0.f;
```

Comment:
so the idea was that when doing a trace for edge walk we use a snap tolerance of 0 because we expected the shape to have the exact points in it from the original route. and to make sure we dont use different points when starting the walk we set a tolerance of 0 so that basically no node snapping occurs.